### PR TITLE
Moonstone 1.0

### LINKED ISSUE #3: Modpack config detection from gradle does not work currently
At the moment the detection of mod loader and minecraft version from a linked gradle project is broken and does not work, which causes these values to always fall back to `Forge/1.16.5` when opening the modlist file.

### PR DESCRIPTION
**Pull request by noeppi-noeppi**
Adds support for Modrinth and changes the `modlist.json` format to store more information. Also renames the package to `org.moddingx.moonstone`.

Includes #4

### CODE REVIEW

**Comment by noeppi-noeppi** [comment]
When this is done, config detection from gradle won't be necessary any longer, so this fixes #3.

**Comment by noeppi-noeppi** [comment]
Actually, it's Moonstone 1.0 as we have not released a 1.0 yet...

File: src/main/scala/org/moddingx/moonstone/model/FileListIO.scala [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import javax.annotation.WillClose
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+object FileListIO {
+  
+  val API: Int = 2
+  
+  private val latestMC = try {
+    val in: Reader = new InputStreamReader(new URL("https://launchermeta.mojang.com/mc/game/version_manifest.json").openStream(), StandardCharsets.UTF_8)
```
Comment (MelanX) [inline src/main/scala/org/moddingx/moonstone/model/FileListIO.scala]: Use `https://piston-meta.mojang.com/mc/game/version_manifest.json`, Mojang changed the location.

**Review by MelanX — CHANGES_REQUESTED** [review]
Only one thing, good luck with it!

**Pull request merged**